NotebookAnalyzers/utils.py

```diff
@@ -27,7 +27,19 @@ def notebookToCode(filename):
     return py_code
 
 def functionsNumber(code):
-    """The function takes a python code sintax tree and returns the number of function definitions"""
+    """The function takes a python code and returns the number of function definitions"""
     tree = ast.parse(code)
     f_num=sum(isinstance(exp, ast.FunctionDef) for exp in tree.body)
-    return f_num
\ No newline at end of file
+    return f_num
+
+def notExecutedCells(notebook):
+    """The function takes the notebook dictionary and returns both the number empty cells in first position and the number of not executed cells in second position"""
+    empty_cells=0
+    not_exec_cells=0
+    for cell in notebook["cells"]:
+        if cell["cell_type"] == 'code':
+            if cell['execution_count']==None and cell['source']==[]:
+                empty_cells=empty_cells+1 #This is an empty Python Cell
+            elif cell['execution_count']==None and cell['source']!=[]:
+                    not_exec_cells=not_exec_cells+1 #This is a not executed Python Cell containing actual code
+    return empty_cells,not_exec_cells
```

Comment:
Qui mi atterrei al single-responsibility principle e splitterei la tua funzione in due funzioni distinte, rispettivamente dedicate alla conta delle celle vuote e alla conta di quelle non eseguite.